hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -575,6 +598,40 @@ private Pair<Integer, HoodieData<HoodieRecord>> initializeRecordIndexPartition()
     return Pair.of(fileGroupCount, records);
   }
 
+  private static HoodieData<HoodieRecord> readRecordKeysFromFileSlices(HoodieEngineContext engineContext,
+                                                                      List<Pair<String, FileSlice>> partitionFileSlicePairs,
```

Comment:
Does this code work for `COW` table too?

Reply:
should work.. For COW, file slice won't have any log files for filegroup id, which is handled by the HoodieMergedLogRecordScanner.

Reply:
yes. It will. Are you suggesting to use the same method for COW as well? 
I am not sure of the perf implications. 
I would say, we need to audit entire code/repo to understand the BaseFile reader code paths and if it can replaced w/ FileSlice reader. 
So that both COW and MOR has same way of reading data files. 

Would prefer to keep it out of the scope of this patch.

Reply:
https://issues.apache.org/jira/browse/HUDI-7756